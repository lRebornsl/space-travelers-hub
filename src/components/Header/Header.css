header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  border-bottom: 1px solid black;
  padding: 1rem 1rem 2rem 1rem;
}

header a {
  text-decoration: none;
}

#logo {
  display: flex;
  align-items: center;
  gap: 10px;
  color: #000;
  font-size: 2rem;
  font-weight: 600;
}

#logo img {
  width: 40px;
  height: 40px;
}