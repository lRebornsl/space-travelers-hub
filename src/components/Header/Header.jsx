import { NavLink } from 'react-router-dom';
import logo from '../../assets/planet_logo.png';
import './Header.css'

const Header = () => {
  return (
    <header className='container'>
      <div>
        <NavLink to="/" id="logo">
          <img src={logo} alt="logo" />
          <h1>Space Travelers&apos; Hub</h1>
        </NavLink>
      </div>
      <nav>
        <ul>
          <li><NavLink to="/myProfile" className="nav_profile">My Profile</NavLink></li>
        </ul>
      </nav>
    </header>
  )
}

export default Header;

