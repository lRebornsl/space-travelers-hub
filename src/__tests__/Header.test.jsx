import { render, waitFor } from '@testing-library/react';
import Header from '../components/Header/Header';
import { beforeEach, describe, it, expect } from 'vitest';
import { Provider } from 'react-redux';
import store from '../redux/Store';
import { BrowserRouter as Router } from 'react-router-dom';

global.mockUseLocation = { pathname: '/' };

describe('Header', () => {
  beforeEach(() => {
    global.useLocation = () => global.mockUseLocation;

    render(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>
    );
  });

  it('Header renders correctly', async () => {
    await waitFor(() => expect(Header).toMatchSnapshot());
  });
});