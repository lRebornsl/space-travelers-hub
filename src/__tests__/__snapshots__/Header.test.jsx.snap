// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > Header renders correctly 1`] = `[Function]`;
